File: valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> check = new Stack<>();
         boolean ret = false;
           for(int i = 0; i < s.length(); i++){
            switch(s.charAt(i)){
                  case '{':
                      check.push('}');
                       break;
                  case '(':
                    check.push(')');
                    break;
                 case '[':
                    check.push(']');
                    break;
                default:
                    if(!check.isEmpty() && check.peek() == s.charAt(i) ){
                        check.pop();
                    }
                    else
                        return false;
                   
            }
        }
        return (check.isEmpty())? true: false;
 }
}

File: After-Camp-LeetCode/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        check = []
        
        for par in s:
            
            if par == '(':
                check.append(")")
            
            elif par == '{':
                check.append("}")
            
            elif par == '[':
                check.append("]")
            
            elif check and check[-1] == par:
                check.pop()
            else:
                return False
        
        return False if check else True